include/glow/Graph/Context.h

```diff
@@ -48,6 +49,9 @@ class Context final {
   /// Maps Placeholder names to Placeholders.
   PlaceholderNameMap nameMap_;
 
+  /// Trace Events recorded during this run.
+  std::vector<TraceEvent> traceEvents_;
```

Comment:
As far as I can see, the intent here is to associate `traceEvents_` with each run.  And `Context` is de-facto a data structure associated with each run.

One problem though is that `Context` is not guaranteed to be associated with each/any run and/or only a single run. 

If we want to extend the semantics of `Context` to guarantee it is associated with a run, we should update the documentation for `Context` and make any such guarantees very explicit there.

Alternatively, we may need to introduce a new data structure to hold the information about runs. This data structure is likely to include `traceEvents_` and probably one or more `Contexts`. This is, of course, something for a separate PR.

Reply:
The context is guaranteed to the be the single per-run object in the Runtime because it must be provided (by unique_ptr) to the DeviceManager while scheduling a new inference - CompiledFunction::execute also requires a Context*. Agree we should update the docs.

It's come up a few times from different folk, do you think there should be a different per run object?

Reply:
Historically, `Context` was introduced just as a simple mapping from placeholders to tensors. Over time, once we introduced the `Runtime` components, we started (mis)using it as a de-facto per run object.

My personal opinion is that our current `Context` is too simple for a per-run object in terms of APIs and information it contains (e.g. a per run object could contain collected information about performance, devices being used, different kinds of counters, mode of execution, etc). So, either we extend `Context` into a per-run object and add more APIs and members as needed and rename it into e.g. a `RunContext` or `ExecutionInfo`; or we should create a new per-run object with all that information.